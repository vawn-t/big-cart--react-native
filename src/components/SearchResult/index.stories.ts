import type { Meta, StoryObj } from '@storybook/react-native';

import SearchResult from '.';
import { Product } from '@models/index';

const meta: Meta<typeof SearchResult> = {
  title: 'Components/SearchResult',
  component: SearchResult,
};

export default meta;

type Story = StoryObj<React.ComponentProps<typeof SearchResult>>;

const DATA: Product[] = [
  {
    id: 1,
    productUnitId: 1,
    name: 'Test',
    image: 'https://clipart-library.com/image_gallery2/Fruit.png',
    quantityStock: 3,
    price: 1,
    categoryId: 2,
  },
  {
    id: 2,
    productUnitId: 1,
    name: 'Test',
    image: 'https://clipart-library.com/image_gallery2/Fruit.png',
    quantityStock: 0,
    price: 1,
    categoryId: 2,
  },
  {
    id: 3,
    productUnitId: 1,
    name: 'Test',
    image: 'https://clipart-library.com/image_gallery2/Fruit.png',
    quantityStock: 3,
    price: 1,
    categoryId: 2,
  },
];

export const Base: Story = {
  args: { results: DATA },
};

export const NotFound: Story = {
  args: {},
};
