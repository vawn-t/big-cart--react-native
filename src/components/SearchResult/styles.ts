import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  notFoundWrapper: {
    paddingVertical: 50,
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export default styles;
