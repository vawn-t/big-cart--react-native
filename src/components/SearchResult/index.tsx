import { memo } from 'react';
import { View } from 'react-native';

// Components
import HorizontalCards from '@components/Cards/HorizontalCards';
import { Typography } from '@components/common';

// Models
import { Product } from '@models/index';

// Types
import { FontWeight, SizeType, TypoVariant } from '@interfaces/typography';

// Utils
import { areEqual } from '@utils/index';

// Styles
import styles from './styles';

interface IProps {
  results: Product[];
}
const SearchResult = ({ results }: IProps) => (
  <View>
    <Typography
      text="Results"
      fontWeight={FontWeight.SemiBold}
      size={SizeType.Large}
      variant={TypoVariant.Paragraph2}
    />

    {results.length ? (
      <HorizontalCards products={results} />
    ) : (
      <View style={styles.notFoundWrapper}>
        <Typography
          text="The products are not available!"
          variant={TypoVariant.Paragraph2}
          fontWeight={FontWeight.SemiBold}
        />
      </View>
    )}
  </View>
);

export default memo(SearchResult, areEqual);
