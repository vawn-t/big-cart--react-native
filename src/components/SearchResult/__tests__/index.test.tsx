import SearchResult from '..';
import { render } from '@utils/index';
import { products } from '@mocks/index';

jest.mock('@react-native-async-storage/async-storage', () =>
  require('@react-native-async-storage/async-storage/jest/async-storage-mock'),
);

describe('SearchResult component', () => {
  const ownProps: React.ComponentProps<typeof SearchResult> = {
    results: products,
  };

  const setup = (overrideProps = {}) => {
    const props = { ...ownProps, ...overrideProps };

    return render(<SearchResult {...props} />);
  };

  it('renders correctly with results', () => {
    const sampleResults = [
      {
        id: 1,
        name: 'Product 1',
        price: 10,
        amount: '2kg',
      },
      {
        id: 2,
        name: 'Product 2',
        price: 15,
        amount: '1kg',
      },
    ];

    const { toJSON } = setup({ results: sampleResults });
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly when no results are available', () => {
    const { toJSON } = setup({ results: [] });
    expect(toJSON()).toMatchSnapshot();
  });
});
