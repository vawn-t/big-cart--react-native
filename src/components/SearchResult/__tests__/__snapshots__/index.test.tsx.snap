// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResult component renders correctly when no results are available 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View>
            <Text
              style={
                [
                  {
                    "color": "#000000",
                  },
                  {
                    "fontFamily": "Poppins-SemiBold",
                  },
                  {
                    "fontSize": 20,
                  },
                  undefined,
                ]
              }
            >
              Results
            </Text>
            <View
              style={
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "paddingVertical": 50,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                    },
                    {
                      "fontFamily": "Poppins-SemiBold",
                    },
                    {
                      "fontSize": 15,
                    },
                    undefined,
                  ]
                }
              >
                The products are not available!
              </Text>
            </View>
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="test"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;

exports[`SearchResult component renders correctly with results 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View>
            <Text
              style={
                [
                  {
                    "color": "#000000",
                  },
                  {
                    "fontFamily": "Poppins-SemiBold",
                  },
                  {
                    "fontSize": 20,
                  },
                  undefined,
                ]
              }
            >
              Results
            </Text>
            <RCTScrollView
              contentContainerStyle={
                {
                  "paddingVertical": 20,
                }
              }
              data={
                [
                  {
                    "amount": "2kg",
                    "id": 1,
                    "name": "Product 1",
                    "price": 10,
                  },
                  {
                    "amount": "1kg",
                    "id": 2,
                    "name": "Product 2",
                    "price": 15,
                  },
                ]
              }
              getItem={[Function]}
              getItemCount={[Function]}
              horizontal={false}
              keyExtractor={[Function]}
              nestedScrollEnabled={true}
              onContentSizeChange={[Function]}
              onLayout={[Function]}
              onMomentumScrollBegin={[Function]}
              onMomentumScrollEnd={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              removeClippedSubviews={false}
              renderItem={[Function]}
              scrollEnabled={false}
              scrollEventThrottle={50}
              stickyHeaderIndices={[]}
              viewabilityConfigCallbackPairs={[]}
            >
              <View>
                <View
                  onFocusCapture={[Function]}
                  onLayout={[Function]}
                  style={null}
                >
                  <View
                    style={
                      [
                        {
                          "flexDirection": "row",
                        },
                        {
                          "alignItems": "center",
                          "columnGap": 20,
                          "flex": 1,
                          "justifyContent": "space-between",
                          "marginVertical": 10,
                        },
                      ]
                    }
                  >
                    <View
                      style={
                        {
                          "backgroundColor": "#FFFFFF",
                          "flex": 0.5,
                        }
                      }
                    >
                      <View
                        style={
                          {
                            "borderBottomWidth": 1,
                            "borderColor": "#EBEBEB",
                            "flex": 5,
                            "justifyContent": "center",
                            "paddingBottom": 10,
                            "paddingTop": 30,
                          }
                        }
                      >
                        <View
                          accessibilityState={
                            {
                              "busy": undefined,
                              "checked": undefined,
                              "disabled": undefined,
                              "expanded": undefined,
                              "selected": undefined,
                            }
                          }
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            {
                              "alignItems": "center",
                            }
                          }
                        >
                          <View
                            style={
                              {
                                "backgroundColor": "#AEDC81",
                                "borderRadius": 42,
                                "height": 84,
                                "position": "absolute",
                                "top": -20,
                                "width": 84,
                              }
                            }
                          />
                          <Image
                            source={
                              {
                                "uri": undefined,
                              }
                            }
                            style={
                              {
                                "height": 84,
                                "objectFit": "contain",
                                "transform": [
                                  {
                                    "scale": 1.2,
                                  },
                                ],
                                "width": 84,
                              }
                            }
                          />
                          <Text
                            style={
                              [
                                {
                                  "color": "#868889",
                                },
                                {
                                  "fontFamily": "Poppins-Medium",
                                },
                                {
                                  "fontSize": 12,
                                },
                                {
                                  "color": "#6CC51D",
                                  "textAlign": "center",
                                },
                              ]
                            }
                          >
                            $10
                          </Text>
                          <Text
                            style={
                              [
                                {
                                  "color": "#000000",
                                },
                                {
                                  "fontFamily": "Poppins-SemiBold",
                                },
                                {
                                  "fontSize": 15,
                                },
                                {
                                  "paddingVertical": 5,
                                  "textAlign": "center",
                                },
                              ]
                            }
                          >
                            Product 1
                          </Text>
                          <Text
                            style={
                              [
                                {
                                  "color": "#868889",
                                },
                                {
                                  "fontFamily": "Poppins-Medium",
                                },
                                {
                                  "fontSize": 12,
                                },
                                {
                                  "textAlign": "center",
                                },
                              ]
                            }
                          />
                        </View>
                      </View>
                      <View
                        style={
                          {
                            "flex": 1,
                            "height": 40,
                            "justifyContent": "center",
                          }
                        }
                      >
                        <View
                          accessibilityState={
                            {
                              "busy": undefined,
                              "checked": undefined,
                              "disabled": undefined,
                              "expanded": undefined,
                              "selected": undefined,
                            }
                          }
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            {
                              "alignItems": "center",
                              "backgroundColor": "#FFFFFF",
                              "flexDirection": "row",
                              "justifyContent": "center",
                            }
                          }
                        >
                          <RNSVGSvgView
                            bbHeight="15"
                            bbWidth="13"
                            color="#6CC51D"
                            fill="none"
                            focusable={false}
                            height={15}
                            style={
                              [
                                {
                                  "backgroundColor": "transparent",
                                  "borderWidth": 0,
                                },
                                {
                                  "flex": 0,
                                  "height": 15,
                                  "width": 13,
                                },
                              ]
                            }
                            tintColor="#6CC51D"
                            width={13}
                          >
                            <RNSVGGroup
                              fill={null}
                              propList={
                                [
                                  "fill",
                                ]
                              }
                            >
                              <RNSVGPath
                                d="m12.997 12.966-.93-9.672c-.02-.214-.216-.377-.445-.377H9.708C9.682 1.304 8.254 0 6.5 0 4.746 0 3.318 1.304 3.292 2.917H1.378c-.232 0-.425.163-.445.377l-.93 9.672c0 .013-.003.025-.003.037C0 14.104 1.093 15 2.438 15h8.124C11.907 15 13 14.104 13 13.003c0-.012 0-.024-.003-.037ZM6.5.828c1.259 0 2.285.933 2.312 2.09H4.188C4.215 1.76 5.241.827 6.5.827Zm4.062 13.344H2.438c-.844 0-1.528-.516-1.541-1.15l.89-9.274h1.501v1.258c0 .23.2.414.449.414s.448-.184.448-.414V3.748h4.627v1.258c0 .23.199.414.448.414.25 0 .448-.184.448-.414V3.748h1.502l.893 9.273c-.013.635-.7 1.15-1.54 1.15Z"
                                fill={
                                  {
                                    "payload": 4285318429,
                                    "type": 0,
                                  }
                                }
                                propList={
                                  [
                                    "fill",
                                  ]
                                }
                              />
                            </RNSVGGroup>
                          </RNSVGSvgView>
                          <Text
                            style={
                              [
                                {
                                  "color": "#000000",
                                },
                                {
                                  "fontFamily": "Poppins-Regular",
                                },
                                {
                                  "fontSize": 15,
                                },
                                {
                                  "paddingLeft": 10,
                                },
                              ]
                            }
                          >
                            Add to cart
                          </Text>
                        </View>
                      </View>
                      <View
                        accessibilityState={
                          {
                            "busy": undefined,
                            "checked": undefined,
                            "disabled": undefined,
                            "expanded": undefined,
                            "selected": undefined,
                          }
                        }
                        accessibilityValue={
                          {
                            "max": undefined,
                            "min": undefined,
                            "now": undefined,
                            "text": undefined,
                          }
                        }
                        accessible={true}
                        collapsable={false}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          {
                            "opacity": 1,
                            "position": "absolute",
                            "right": 10,
                            "top": 10,
                          }
                        }
                      >
                        <RNSVGSvgView
                          bbHeight="18"
                          bbWidth="22"
                          fill="none"
                          focusable={false}
                          height={18}
                          style={
                            [
                              {
                                "backgroundColor": "transparent",
                                "borderWidth": 0,
                              },
                              {
                                "flex": 0,
                                "height": 18,
                                "width": 22,
                              },
                            ]
                          }
                          width={22}
                        >
                          <RNSVGGroup
                            fill={null}
                            propList={
                              [
                                "fill",
                              ]
                            }
                          >
                            <RNSVGPath
                              d="M11 18c-.313 0-.615-.104-.85-.294a249.404 249.404 0 0 0-2.502-1.979l-.004-.003c-2.216-1.737-4.13-3.237-5.462-4.714C.693 9.358 0 7.79 0 6.08c0-1.662.62-3.196 1.745-4.318C2.885.626 4.447 0 6.145 0c1.27 0 2.434.37 3.458 1.097.516.368.984.818 1.397 1.342.413-.524.88-.974 1.398-1.342A5.849 5.849 0 0 1 15.854 0c1.7 0 3.262.626 4.4 1.762C21.38 2.884 22 4.418 22 6.08c0 1.711-.693 3.278-2.182 4.93-1.332 1.477-3.245 2.977-5.461 4.714-.758.593-1.616 1.266-2.507 1.982-.235.19-.537.294-.85.294ZM6.146 1.185c-1.335 0-2.561.49-3.453 1.38-.906.903-1.404 2.151-1.404 3.515 0 1.44.581 2.726 1.885 4.173 1.26 1.399 3.135 2.868 5.306 4.57l.004.002c.759.595 1.619 1.27 2.514 1.99.9-.722 1.762-1.397 2.523-1.993 2.17-1.701 4.045-3.17 5.305-4.569 1.304-1.447 1.885-2.734 1.885-4.173 0-1.364-.498-2.612-1.404-3.515-.891-.89-2.118-1.38-3.453-1.38-.978 0-1.875.286-2.668.85-.707.502-1.199 1.138-1.487 1.582a.823.823 0 0 1-.699.365.823.823 0 0 1-.699-.365c-.288-.444-.78-1.08-1.487-1.582a4.517 4.517 0 0 0-2.668-.85Z"
                              fill={
                                {
                                  "payload": 4287006857,
                                  "type": 0,
                                }
                              }
                              propList={
                                [
                                  "fill",
                                ]
                              }
                            />
                          </RNSVGGroup>
                        </RNSVGSvgView>
                      </View>
                    </View>
                    <View
                      style={
                        {
                          "backgroundColor": "#FFFFFF",
                          "flex": 0.5,
                        }
                      }
                    >
                      <View
                        style={
                          {
                            "borderBottomWidth": 1,
                            "borderColor": "#EBEBEB",
                            "flex": 5,
                            "justifyContent": "center",
                            "paddingBottom": 10,
                            "paddingTop": 30,
                          }
                        }
                      >
                        <View
                          accessibilityState={
                            {
                              "busy": undefined,
                              "checked": undefined,
                              "disabled": undefined,
                              "expanded": undefined,
                              "selected": undefined,
                            }
                          }
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            {
                              "alignItems": "center",
                            }
                          }
                        >
                          <View
                            style={
                              {
                                "backgroundColor": "#AEDC81",
                                "borderRadius": 42,
                                "height": 84,
                                "position": "absolute",
                                "top": -20,
                                "width": 84,
                              }
                            }
                          />
                          <Image
                            source={
                              {
                                "uri": undefined,
                              }
                            }
                            style={
                              {
                                "height": 84,
                                "objectFit": "contain",
                                "transform": [
                                  {
                                    "scale": 1.2,
                                  },
                                ],
                                "width": 84,
                              }
                            }
                          />
                          <Text
                            style={
                              [
                                {
                                  "color": "#868889",
                                },
                                {
                                  "fontFamily": "Poppins-Medium",
                                },
                                {
                                  "fontSize": 12,
                                },
                                {
                                  "color": "#6CC51D",
                                  "textAlign": "center",
                                },
                              ]
                            }
                          >
                            $15
                          </Text>
                          <Text
                            style={
                              [
                                {
                                  "color": "#000000",
                                },
                                {
                                  "fontFamily": "Poppins-SemiBold",
                                },
                                {
                                  "fontSize": 15,
                                },
                                {
                                  "paddingVertical": 5,
                                  "textAlign": "center",
                                },
                              ]
                            }
                          >
                            Product 2
                          </Text>
                          <Text
                            style={
                              [
                                {
                                  "color": "#868889",
                                },
                                {
                                  "fontFamily": "Poppins-Medium",
                                },
                                {
                                  "fontSize": 12,
                                },
                                {
                                  "textAlign": "center",
                                },
                              ]
                            }
                          />
                        </View>
                      </View>
                      <View
                        style={
                          {
                            "flex": 1,
                            "height": 40,
                            "justifyContent": "center",
                          }
                        }
                      >
                        <View
                          accessibilityState={
                            {
                              "busy": undefined,
                              "checked": undefined,
                              "disabled": undefined,
                              "expanded": undefined,
                              "selected": undefined,
                            }
                          }
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            {
                              "alignItems": "center",
                              "backgroundColor": "#FFFFFF",
                              "flexDirection": "row",
                              "justifyContent": "center",
                            }
                          }
                        >
                          <RNSVGSvgView
                            bbHeight="15"
                            bbWidth="13"
                            color="#6CC51D"
                            fill="none"
                            focusable={false}
                            height={15}
                            style={
                              [
                                {
                                  "backgroundColor": "transparent",
                                  "borderWidth": 0,
                                },
                                {
                                  "flex": 0,
                                  "height": 15,
                                  "width": 13,
                                },
                              ]
                            }
                            tintColor="#6CC51D"
                            width={13}
                          >
                            <RNSVGGroup
                              fill={null}
                              propList={
                                [
                                  "fill",
                                ]
                              }
                            >
                              <RNSVGPath
                                d="m12.997 12.966-.93-9.672c-.02-.214-.216-.377-.445-.377H9.708C9.682 1.304 8.254 0 6.5 0 4.746 0 3.318 1.304 3.292 2.917H1.378c-.232 0-.425.163-.445.377l-.93 9.672c0 .013-.003.025-.003.037C0 14.104 1.093 15 2.438 15h8.124C11.907 15 13 14.104 13 13.003c0-.012 0-.024-.003-.037ZM6.5.828c1.259 0 2.285.933 2.312 2.09H4.188C4.215 1.76 5.241.827 6.5.827Zm4.062 13.344H2.438c-.844 0-1.528-.516-1.541-1.15l.89-9.274h1.501v1.258c0 .23.2.414.449.414s.448-.184.448-.414V3.748h4.627v1.258c0 .23.199.414.448.414.25 0 .448-.184.448-.414V3.748h1.502l.893 9.273c-.013.635-.7 1.15-1.54 1.15Z"
                                fill={
                                  {
                                    "payload": 4285318429,
                                    "type": 0,
                                  }
                                }
                                propList={
                                  [
                                    "fill",
                                  ]
                                }
                              />
                            </RNSVGGroup>
                          </RNSVGSvgView>
                          <Text
                            style={
                              [
                                {
                                  "color": "#000000",
                                },
                                {
                                  "fontFamily": "Poppins-Regular",
                                },
                                {
                                  "fontSize": 15,
                                },
                                {
                                  "paddingLeft": 10,
                                },
                              ]
                            }
                          >
                            Add to cart
                          </Text>
                        </View>
                      </View>
                      <View
                        accessibilityState={
                          {
                            "busy": undefined,
                            "checked": undefined,
                            "disabled": undefined,
                            "expanded": undefined,
                            "selected": undefined,
                          }
                        }
                        accessibilityValue={
                          {
                            "max": undefined,
                            "min": undefined,
                            "now": undefined,
                            "text": undefined,
                          }
                        }
                        accessible={true}
                        collapsable={false}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          {
                            "opacity": 1,
                            "position": "absolute",
                            "right": 10,
                            "top": 10,
                          }
                        }
                      >
                        <RNSVGSvgView
                          bbHeight="18"
                          bbWidth="22"
                          fill="none"
                          focusable={false}
                          height={18}
                          style={
                            [
                              {
                                "backgroundColor": "transparent",
                                "borderWidth": 0,
                              },
                              {
                                "flex": 0,
                                "height": 18,
                                "width": 22,
                              },
                            ]
                          }
                          width={22}
                        >
                          <RNSVGGroup
                            fill={null}
                            propList={
                              [
                                "fill",
                              ]
                            }
                          >
                            <RNSVGPath
                              d="M11 18c-.313 0-.615-.104-.85-.294a249.404 249.404 0 0 0-2.502-1.979l-.004-.003c-2.216-1.737-4.13-3.237-5.462-4.714C.693 9.358 0 7.79 0 6.08c0-1.662.62-3.196 1.745-4.318C2.885.626 4.447 0 6.145 0c1.27 0 2.434.37 3.458 1.097.516.368.984.818 1.397 1.342.413-.524.88-.974 1.398-1.342A5.849 5.849 0 0 1 15.854 0c1.7 0 3.262.626 4.4 1.762C21.38 2.884 22 4.418 22 6.08c0 1.711-.693 3.278-2.182 4.93-1.332 1.477-3.245 2.977-5.461 4.714-.758.593-1.616 1.266-2.507 1.982-.235.19-.537.294-.85.294ZM6.146 1.185c-1.335 0-2.561.49-3.453 1.38-.906.903-1.404 2.151-1.404 3.515 0 1.44.581 2.726 1.885 4.173 1.26 1.399 3.135 2.868 5.306 4.57l.004.002c.759.595 1.619 1.27 2.514 1.99.9-.722 1.762-1.397 2.523-1.993 2.17-1.701 4.045-3.17 5.305-4.569 1.304-1.447 1.885-2.734 1.885-4.173 0-1.364-.498-2.612-1.404-3.515-.891-.89-2.118-1.38-3.453-1.38-.978 0-1.875.286-2.668.85-.707.502-1.199 1.138-1.487 1.582a.823.823 0 0 1-.699.365.823.823 0 0 1-.699-.365c-.288-.444-.78-1.08-1.487-1.582a4.517 4.517 0 0 0-2.668-.85Z"
                              fill={
                                {
                                  "payload": 4287006857,
                                  "type": 0,
                                }
                              }
                              propList={
                                [
                                  "fill",
                                ]
                              }
                            />
                          </RNSVGGroup>
                        </RNSVGSvgView>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </RCTScrollView>
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="test"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
